### PR TITLE
FormatCommand.formatGenerated allows ignoring generated files (.g.dart)

### PR DESCRIPTION
**Pull request by passsy**
```dart
// ignores *.g.dart and *.freezed.dart files
runner.addCommand(FormatCommand(formatGenerated: false));
```

### CODE REVIEW

File: sidekick_core/lib/src/commands/format_command.dart [hunk 1]
```diff
@@ -96,17 +103,18 @@ class FormatCommand extends Command {
 
     final allFiles =
         SidekickContext.projectRoot.findFilesToFormat(globExcludes).toList();
+    final withoutGenerated = filterGeneratedFiles(allFiles).toList();
```
Comment (nohli) [inline sidekick_core/lib/src/commands/format_command.dart:106]: Is this filename still correct?
If formatGenerated is true, this includes all files.

Comment (passsy) [reply sidekick_core/lib/src/commands/format_command.dart:106]: Any better idea than `maybeWithithOrWithoutGeneratedFiles`?

Comment (nohli) [reply sidekick_core/lib/src/commands/format_command.dart:106]: filteredFiles?

**Review by Giuspepe — APPROVED** [review]
LGTM but I think we can do this with much less code

```dart
  final bool formatGenerated;

  FormatCommand({
    List<String> excludeGlob = const [],
    this.defaultLineLength = 80,
    this.formatGenerated = true,
  }) : this.excludeGlob = [
          ...excludeGlob,
          if (!formatGenerated) ...['**.freezed.dart', '**.g.dart'],
        ] {
    ...
  }
```

**Pull request merged**